homeassistant/components/synology_dsm/camera.py

```diff
@@ -64,16 +64,15 @@ def __init__(
         self,
         api: SynoApi,
         coordinator: SynologyDSMCameraUpdateCoordinator,
-        camera_id: str,
+        camera_id: int,
     ) -> None:
         """Initialize a Synology camera."""
+        camera = coordinator.data["cameras"][camera_id]
         description = SynologyDSMCameraEntityDescription(
             api_key=SynoSurveillanceStation.CAMERA_API_KEY,
-            key=camera_id,
-            name=coordinator.data["cameras"][camera_id].name,
-            entity_registry_enabled_default=coordinator.data["cameras"][
-                camera_id
-            ].is_enabled,
+            key=str(camera_id),
```

Comment:
This is kind misusing this parameter to transfer an device identifier (which is not what this key is meant for).

It now is converted to a string, and later back to an integer again (which is just unneeded and unlogical).

Let's not use the key of a entity description for that.